src/silverlabnwb/nwb_file.py

```diff
@@ -375,7 +379,7 @@ def add_time_series_data(self, label, data, times, ts_attrs={}, data_attrs={},
         """
         all_attrs = dict(ts_attrs)
         all_attrs.update(data_attrs)
-        if compress:
+        if self.compress and kind is not ImageSeries:
```

Comment:
The intention for this is to avoid compressing the video `ImageSeries`, which has no data and would give an error. We may end up including another `ImageSeries` in the future, so it's probably better to check that `data is not None` rather than the class.